cl/search/views.py

```diff
@@ -514,41 +519,59 @@ def show_results(request: HttpRequest) -> HttpResponse:
                 if not is_bot(request):
                     async_to_sync(tally_stat)("search.results")
 
+                    # Perform the search
+                    search_type = request.GET.get("type", SEARCH_TYPES.OPINION)
+
+                    if search_type == SEARCH_TYPES.PARENTHETICAL:
+                        render_dict.update(do_es_search(request.GET.copy()))
+                    elif search_type == SEARCH_TYPES.ORAL_ARGUMENT:
+                        # Check if waffle flag is active.
+                        if waffle.flag_is_active(request, "oa-es-active"):
+                            render_dict.update(
+                                do_es_search(request.GET.copy())
+                            )
+                        else:
+                            render_dict.update(do_search(request.GET.copy()))
+                    elif search_type == SEARCH_TYPES.PEOPLE:
+                        if waffle.flag_is_active(request, "p-es-active"):
+                            render_dict.update(
+                                do_es_search(request.GET.copy())
+                            )
+                        else:
+                            render_dict.update(do_search(request.GET.copy()))
+                    elif search_type in [
+                        SEARCH_TYPES.RECAP,
+                        SEARCH_TYPES.DOCKETS,
+                    ]:
+                        if waffle.flag_is_active(request, "r-es-active"):
+                            search_results = do_es_search(request.GET.copy())
+                        else:
+                            search_results = do_search(request.GET.copy())
+                        render_dict.update(search_results)
+                    elif search_type == SEARCH_TYPES.OPINION:
+                        if waffle.flag_is_active(request, "o-es-active"):
+                            render_dict.update(
+                                do_es_search(request.GET.copy())
+                            )
+                        else:
+                            render_dict.update(do_search(request.GET.copy()))
+                    elif search_type == SEARCH_TYPES.RECAP_DOCUMENT:
+                        render_dict.update(do_es_search(request.GET.copy()))
+                    else:
+                        render_dict.update(do_search(request.GET.copy()))
+
+                    # Create and save the SearchQuery object
+                    SearchQuery.objects.create(
+                        get_params=request.GET.urlencode(),
+                        query_time_ms=render_dict.get("query_time", 0),
```

Comment:
I think I'd set this to null if we don't have the value, or if the called function always returns the value, I'd do:

```suggestion
                        query_time_ms=render_dict["query_time"],
```

By doing that, you'd enforce the calling function to always deliver on its promise.